app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -368,28 +369,29 @@ trait TransactionQueries extends StrictLogging {
                   gasPrice)
     }
 
-  def getBalanceQuerySQL(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
+  def getBalanceQueryDEPRECATED(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
     sql"""
         SELECT outputs.amount, outputs.lock_time
         FROM outputs
-        LEFT JOIN inputs ON outputs.key = inputs.output_ref_key
-        WHERE outputs.main_chain = true AND outputs.address = $address AND inputs.block_hash IS NULL
+        LEFT JOIN inputs
+        ON outputs.key = inputs.output_ref_key
+        WHERE outputs.main_chain = true
+        AND outputs.address = $address
+        AND inputs.block_hash IS NULL
       """.as[(U256, Option[TimeStamp])]
   }
 
-  val getBalanceQuery = Compiled { address: Rep[Address] =>
-    mainOutputs
-      .filter(output => output.address === address)
-      .joinLeft(mainInputs)
-      .on(_.key === _.outputRefKey)
-      .filter(_._2.isEmpty)
-      .map { case (output, _) => (output.amount, output.lockTime) }
-  }
-
-  def getBalanceAction(address: Address): DBActionR[(U256, U256)] = {
-    getBalanceQuery(address).result.map { outputs =>
-      sumBalance(outputs)
-    }
+  def getBalanceQuery(address: Address): DBActionSR[(U256, Option[TimeStamp])] = {
+    sql"""
+      SELECT outputs.amount, outputs.lock_time
+      FROM outputs
+      LEFT JOIN inputs
+      ON outputs.key = inputs.output_ref_key
+      WHERE outputs.spent_finalized IS NULL
+      AND outputs.main_chain = true
+      AND outputs.address = $address
+      AND inputs.block_hash IS NULL;
+    """.as[(U256, Option[TimeStamp])]
```

Comment:
@simerplaha Could you help us check if this query is using the partial index `createNonSpentIndex`?

Reply:
![maim-1650537499](https://user-images.githubusercontent.com/2979182/164441089-ce33c58e-4ea4-4148-83b8-82d1ec3133e3.png)

With `EXPLAIN` we see: `using non_spent_output_idx on outputs`

Reply:
As I comment in another thread, I got why the index is used even if we don't use `block_hash` here. The mechanism is more complicated than I thought, and the knowledge is very necessary for proper optimization. It turns out multi-column indexing is not that needed for output schema in this case as far as I see.